File: codes/Summary_Ranges.java
// 这题涉及到一些字符串的操作，不想用c++来做
// 思路很简单

public class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList<String>();
        
        int length = nums.length;
        if (length == 0)
            return result;
            
        int first = nums[0];
        String tmp = first + "";
        for (int i = 1; i < length; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                tmp = first + "->" + nums[i];
            } else {
                result.add(tmp);
                first = nums[i];
                tmp = first + "";
            }
        }
        result.add(tmp);
        
        return result;
    } 
}

File: codes/Summary_Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        for (int i = 0; i < nums.size(); ) {
            int cur = i;
            while (cur + 1 < nums.size() && nums[cur] + 1 == nums[cur + 1]) cur++;
            if (i == cur) {
                result.push_back(to_string(nums[i]));
            } else {
                string tmp = to_string(nums[i]) + "->" + to_string(nums[cur]);
                result.push_back(tmp);
            }
            i = cur + 1;
        }
        return result;
    }
};
